internal/pkg/cli/init.go

```diff
@@ -466,16 +475,20 @@ func (o *initOpts) deployEnv() error {
 	if initEnvCmd, ok := o.initEnvCmd.(*initEnvOpts); ok {
 		// Set the application name from app init to the env init command.
 		initEnvCmd.appName = *o.appName
+		initEnvCmd.name = *o.envName
```

Comment:
since this happens, could we just remove L147? since L147 only works for flags, but this case happens for flags _and_ typed in env names

Reply:
same goes for `deployEnvCmd`, `deploySvcCmd`, and `deployJobCmd`

Reply:
just trying to avoid us missing bugs bc we only use flags in our e2e tests (previous case https://github.com/aws/copilot-cli/pull/5126)